examples/interpreting_monte_carlo_tree_search.ipynb

```diff
@@ -0,0 +1,1100 @@
+    "    min_atoms : int\n",
+    "        The minimum number of atoms in a subgraph.\n",
+    "    c_puct : float\n",
+    "        The constant controlling the level of exploration.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    list\n",
+    "        A list of rationales each represented by a MCTSNode.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    mol = Chem.MolFromSmiles(smiles)\n",
+    "\n",
+    "    clusters, atom_cls = find_clusters(mol)\n",
+    "    nei_cls = [0] * len(clusters)\n",
+    "    for i, cls in enumerate(clusters):\n",
+    "        nei_cls[i] = [nei for atom in cls for nei in atom_cls[atom]]\n",
+    "        nei_cls[i] = set(nei_cls[i]) - {i}\n",
+    "        clusters[i] = set(list(cls))\n",
+    "    for a in range(len(atom_cls)):\n",
+    "        atom_cls[a] = set(atom_cls[a])\n",
+    "\n",
+    "    root = MCTSNode(smiles, set(range(mol.GetNumAtoms())))\n",
+    "    state_map = {smiles: root}\n",
+    "    for _ in range(n_rollout):\n",
+    "        mcts_rollout(\n",
+    "            root,\n",
+    "            state_map,\n",
+    "            smiles,\n",
+    "            clusters,\n",
+    "            atom_cls,\n",
+    "            nei_cls,\n",
+    "            scoring_function,\n",
+    "            min_atoms=min_atoms,\n",
+    "            c_puct=c_puct,\n",
+    "        )\n",
+    "\n",
+    "    rationales = [\n",
+    "        node\n",
+    "        for _, node in state_map.items()\n",
+    "        if len(node.atoms) <= max_atoms and node.P >= prop_delta\n",
+    "    ]\n",
+    "\n",
+    "    return rationales"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Running interpretation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# MCTS options\n",
+    "property_for_interpretation = \"lipo\"\n",
+    "\n",
+    "rollout = 10  # number of MCTS rollouts to perform. If mol.GetNumAtoms() > 50, consider setting n_rollout = 1 to avoid long computation time\n",
+    "\n",
+    "c_puct = 10.0  # constant that controls the level of exploration\n",
+    "\n",
+    "max_atoms = 20  # maximum number of atoms allowed in an extracted rationale\n",
+    "\n",
+    "min_atoms = 8  # minimum number of atoms in an extracted rationale\n",
+    "\n",
+    "prop_delta = 0.5  # Minimum score to count as positive.\n",
+    "# In this algorithm, if the predicted property from the substructure if larger than prop_delta, the substructure is considered satisfactory.\n",
+    "# This value depends on the property you want to interpret. 0.5 is a dummy value for demonstration purposes\n",
+    "\n",
+    "num_rationales_to_keep = 5  # number of rationales to keep for each molecule\n",
+    "\n",
+    "# Prediction options\n",
+    "batch_size = 64  # batch size for making predictions\n",
+    "num_workers = 0  # number of workers for making predictions"
```

Comment:
Also related is the warning about dropping the last batch. You loop through smiles one at a time `for smiles in all_smiles:` so every batch will always have only one molecule regardless of `batch_size`. `trainer.predict` will automatically turn off `drop_last` so your code works, but you could turn off the warning by hard coding `batch_size` to be 1.

Reply:
Thanks. I just make the batch_size to be hard coded as 1 and num_workers to be hard coded as 0, because the current set up of mcts cannot perform tree search for multiple molecules at the same time.